android/src/main/java/com/github/yamill/orientation/listeners/OrientationConfigListener.kt

```diff
@@ -12,21 +12,19 @@ import com.facebook.react.bridge.LifecycleEventListener
 import com.facebook.react.bridge.ReactContext
 import com.facebook.react.common.ReactConstants
 import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.github.yamill.orientation.util.OrientationUtil
 
 class OrientationConfigListener internal constructor(
-    reactContext: ReactContext,
+    private val reactContext: ReactContext,
     private val onGetCurrentActivity: () -> Activity?,
 ) : LifecycleEventListener {
 
     private val receiver = object : BroadcastReceiver() {
 
         override fun onReceive(context: Context, intent: Intent) {
             val newConfig = intent.getParcelableExtra<Configuration>(INTENT_VALUE_KEY)!!
-            val orientationValue = if (newConfig.orientation == 1) {
-                "PORTRAIT"
-            } else {
-                "LANDSCAPE"
-            }
+            val orientationValue = OrientationUtil.getOrientationString(newConfig.orientation)
+            tryEmitOrientationChange(orientationValue, reactContext)
```

Comment:
Since you're now calling `tryEmitOrientationChange`, do you also need to delete the

```
val params = Arguments.createMap()
            params.putString("orientation", orientationValue)
            if (reactContext.hasActiveReactInstance()) {
                reactContext
                    .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
                    .emit("orientationDidChange", params)
            }
```

part below? Otherwise, it looks like it'll emit the same change twice in a row

Reply:
Ouch, good catch.